src/univariate/continuous/ekdist.jl

```diff
@@ -0,0 +1,224 @@
+    u::T
+    EKDist{T}(α, β, γ, l, u) where {T} = new{T}(α, β, γ, l, u) 
+end
+
+function EKDist(α::T, β::T, γ::T, l::T, u::T; check_args::Bool=true) where {T <: Real}
+    @check_args EKDist (α, α ≥ zero(α)) (β, β ≥ zero(β)) (γ, γ ≥ zero(γ)) (l < u)
+    return EKDist{T}(α, β, γ, l, u) 
+end
+
+EKDist(α::Real, β::Real, γ::Real, l::Real, u::Real; check_args::Bool=true) = 
+    EKDist(promote(α, β, γ, l, u)...; check_args=check_args)
+
+EKDist(α::Integer, β::Integer, γ::Integer, l::Integer, u::Integer ; check_args::Bool=true) = 
+    EKDist(float(α), float(β), float(γ), float(l), float(u); check_args=check_args)
+
+EKDist(α::Real, β::Real, γ::Real; check_args::Bool=true) = EKDist(promote(α, β, γ, 0.0, 1.0)...; check_args=check_args)
+EKDist(α::Real, β::Real; check_args::Bool=true) = EKDist(promote(α, β, 1.0, 0.0, 1.0)...; check_args=check_args)
+
+# @distr_support macro would set constant bounds but we need to take bounds from input
+# so we will just define them ourselves
+maximum(d::EKDist) = d.u
+minimum(d::EKDist) = d.l
+
+#### Conversions
+convert(::Type{EKDist{T}}, α::S, β::S, γ::S, l::S, u::S) where {T <: Real, S <: Real} = 
+        EKDist(T(α), T(β), T(γ), T(l), T(u))
+        
+convert(::Type{EKDist{T}}, d::EKDist) where {T<:Real} = 
+        EKDist{T}(T(d.α), T(d.β), T(d.γ), T(d.l), T(d.u))
+
+convert(::Type{EKDist{T}}, d::EKDist{T}) where {T<:Real} = d
+
+#### Parameters
+
+params(d::EKDist) = (d.α, d.β, d.γ, d.l, d.u)
+partype(::EKDist{T}) where {T} = T
+
+#### Evaluation
+
+function pdf(d::EKDist, x::Real)
+      xs = (x - d.l) / (d.u - d.l)
+      α = d.α
+      β = d.β
+      γ = d.γ
+    return ( α * β * γ * 
+             xs^(α - 1) * 
+             (1 - xs^α)^(β - 1) * 
+             (1 - (1 - xs^α)^β)^(γ - 1)
+           )
+end
+
+function logpdf(d::EKDist, x::Real)
+      xs =  (x - d.l) / (d.u - d.l) 
+      α = d.α
+      β = d.β
+      γ = d.γ
+    return  ( log(α) + log(β) + log(γ) + 
+              log(xs)*(α - 1) + 
+              log(1 - xs^α)*(β - 1) + 
+              log(1 - (1 - xs^α)^β)*(γ - 1)
+            ) 
+end
+
+function logpdf(d::EKDist, x::Real, edgebound::Real)
+    # Failes at exactly 1 or 0, so I guess jitter it by small amount
+    # this only matters to me in likelihood analysis for real data
+    # that might get defined as the max of the data or Fitting
+    # bound parameters as part of the process.
+    xc = clamp( x, d.l + edgebound, d.u - edgebound)
+    return logpdf(d, xc)  
+end
+
+# do we even need gradlogpdf?
+# grad implied gradient but this is definitely derivative wrt x
+function gradlogpdf(d::EKDist, x::Real)
+    outofsupport = (x < d.l) | (x > d.u)
+    xs = (x - d.l) / (d.u - d.l)
+    α = d.α
+    β = d.β
+    γ = d.γ
+
+    z = (α - 1) / xs +
+        -(α * xs^(α - 1)) / (1 - xs^α) * (β - 1) +
+        (α * β * xs^(α - 1)) * (1 - xs^α)^(β - 1) / (1 -(1 - xs^α)^β) * (γ - 1)
+    
+    return outofsupport ? zero(z) : z
+end
+
+function gradlogpdf(d::EKDist, x::Real, edgebound::Real)
```

Comment:
The `edgebound` stuff shouldn't be necessary. What is the problem you ran into that necessitated it?